src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -45,10 +47,11 @@ class InstanceOpFactoryImpl(
   }
 
   private[this] def inferNormalTaskOp(request: InstanceOpFactory.Request): Option[InstanceOp] = {
-    val InstanceOpFactory.Request(runSpec, offer, tasks, _) = request
+    val InstanceOpFactory.Request(runSpec, offer, instances, _) = request
+    val tasks: Seq[Task] = instances.values.flatMap(_.tasks)(collection.breakOut)
 
     new TaskBuilder(runSpec, Task.Id.forRunSpec, config, Some(appTaskProc)).
-      buildIfMatches(offer, tasks.values.flatMap(_.tasks)).map {
+      buildIfMatches(offer, tasks).map {
```

Comment:
there's a whole new builder coming for building pods from task-groups

On Tue, Sep 13, 2016 at 11:44 AM, Matthias Eichstedt <
notifications@github.com> wrote:

> In src/main/scala/mesosphere/marathon/core/launcher/impl/
> InstanceOpFactoryImpl.scala
> <https://github.com/mesosphere/marathon/pull/4347#discussion_r78586578>:
>
> >
> >      new TaskBuilder(runSpec, Task.Id.forRunSpec, config, Some(appTaskProc)).
> > -      buildIfMatches(offer, tasks.values.flatMap(_.tasks)).map {
> > +      buildIfMatches(offer, tasks).map {
>
> In order to match constraints for pods we should rather be passing
> instances instead of tasks, but before that's adjusted this is OK
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4347/files/bf792261a825b89046d4d7e4de2aa3b2817b788d#r78586578>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLKdqs-qQMsbMfxnzF2qhMSbluaZRks5qpsTTgaJpZM4J5a4D>
> .
>